prober/notification.go

```diff
@@ -0,0 +1,139 @@
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+var (
+	alertGenerated   = expvar.NewInt("alert_generated")
+	alertFailed      = expvar.NewInt("alert_failed")
+	warningGenerated = expvar.NewInt("warning_generated")
+	warningFailed    = expvar.NewInt("warning_failed")
+)
+
+// sendAlert sends an alert to the incident response system, to
+// page a human responder immediately.
+// summary should be short and state the nature of the emergency.
+// details can be longer, up to 29 KBytes.
+func sendAlert(summary, details string) error {
+	type squadcastAlert struct {
+		Message     string            `json:"message"`
+		Description string            `json:"description"`
+		Tags        map[string]string `json:"tags,omitempty"`
+		Status      string            `json:"status"`
+		EventId     string            `json:"event_id"`
+	}
+
+	tags := make(map[string]string)
+	tags["severity"] = "critical"
+	sqa := squadcastAlert{
+		Message:     summary,
+		Description: details,
+		Tags:        tags,
+		Status:      "trigger",
+		EventId:     uuid.New().String(),
+	}
+
+	sqBody, err := json.Marshal(sqa)
+	if err != nil {
+		alertFailed.Add(1)
+		return err
+	}
+
+	webhookUrl := os.Getenv("SQUADCAST_WEBHOOK")
+	if webhookUrl == "" {
+		warningFailed.Add(1)
+		return errors.New("No SQUADCAST_WEBHOOK configured")
+	}
+
+	req, err := http.NewRequest(http.MethodPost, webhookUrl, bytes.NewBuffer(sqBody))
+	if err != nil {
+		alertFailed.Add(1)
+		return err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		alertFailed.Add(1)
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		alertFailed.Add(1)
+		return errors.New(resp.Status)
+	}
+
+	body, _ := io.ReadAll(resp.Body)
+	if string(body) != "ok" {
+		alertFailed.Add(1)
+		return errors.New("Non-ok response returned from Squadcast")
+	}
+
+	alertGenerated.Add(1)
+	return nil
+}
+
+// sendWarning will post a message to Slack.
+// details should be a description of the issue.
+func sendWarning(details string) error {
+	type SlackRequestBody struct {
+		Text string `json:"text"`
+	}
+
+	slackBody, err := json.Marshal(SlackRequestBody{Text: details})
+	if err != nil {
+		warningFailed.Add(1)
+		return err
+	}
+
+	webhookUrl := os.Getenv("SLACK_WEBHOOK")
+	if webhookUrl == "" {
+		warningFailed.Add(1)
+		return errors.New("No SLACK_WEBHOOK configured")
+	}
+
+	req, err := http.NewRequest("POST", webhookUrl, bytes.NewReader(slackBody))
+	if err != nil {
+		warningFailed.Add(1)
+		return err
+	}
+	req.Header.Add("Content-Type", "application/json")
+
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		warningFailed.Add(1)
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		warningFailed.Add(1)
+		return errors.New(resp.Status)
+	}
+
+	body, _ := io.ReadAll(resp.Body)
+	if string(body) != "ok" {
```

Comment:
Is there any chance there is extra whitespace in the reply? Maybe
```suggestion
	if s := strings.TrimSpace(string(body)); s != "ok" {
```
?